futures-util/src/io/mod.rs

```diff
@@ -288,7 +288,7 @@ pub trait AsyncWriteExt: AsyncWrite {
     /// assert_eq!(output, [1, 2, 3, 4, 0]);
     /// # Ok::<(), Box<std::error::Error>>(()) }).unwrap();
     /// ```
-    fn write_all<'a>(&'a mut self, buf: &'a [u8]) -> WriteAll<'a, Self> {
+    fn write_all(&'a mut self, buf: &'a [u8]) -> WriteAll<'a, Self> {
```

Comment:
Was it intentional to use in-band-lifetimes here?

Reply:
Yep, as far as I’m aware they’re still on track to be stable in 2018 Edition. If you’ve got some reason not to use them I can revert these changes.

Reply:
@cramertj My local waker opt PR you just merged also [had an in-band lifetime](https://github.com/rust-lang-nursery/futures-rs/blob/476fc37b05c9387d8d8604e03f3a11aea1ec0f3f/futures-util/src/stream/futures_unordered/node.rs#L101) in it 😇